loopy/symbolic.py

```diff
@@ -758,6 +784,25 @@ def copy(self, *, name=None, tags=None):
         return TaggedVariable(name, tags)
 
 
+@p.expr_dataclass()
+class TaggedExpression(LoopyExpressionBase):
+    """
+    Represents a frozenset of tags attached to an :attr:`expr`.
+
+    .. attribute:: tags
+
+        A :class:`frozenset` of subclasses of :class:`pytools.tag.Tag` used to
+        provide metadata on this expression.
+
+    .. attribute:: expr
+
+        An expression to which :attr:`tags` are attached.
+    """
+
+    tags: frozenset[Tag]
+    expr: Expression
```

Comment:
Have this use `autoattribute`, move docstrings after declaration.

Reply:
Done in 6661eff1